compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Optics.hs

```diff
@@ -164,3 +173,100 @@ exprValueRef f = cata go
     go = \case
       EValF val -> EVal <$> f val
       e -> embed <$> sequenceA e
+
+class HasPackageRefs' a where
```

Comment:
@hurryabit would be great to get your view if this is a reasonable approach. Seems fine to me.